### PR TITLE
Updated to document breaking changes

### PR DESCRIPTION
**Pull request by MaryanneNjeri**
This PR updates the AzureDevOpsPushPipelineExtenstion.md and documents importing feature flag breaking change 
GitHub issue #760

### CODE REVIEW

File: releaseNotes/AzureDevOpsPushPipelineExtension.md [hunk 1]
```diff
@@ -3,7 +3,15 @@ Azure App Configuration Push extension for Azure DevOps pipeline can be installe
 ![sample](pictures/AzureDevOpsPushExtensionVersionSample.PNG)
 
 ### v5.0.0 - February, 02 2023
-Updated the task to require Node.js 16. It previously required 10
+* Updated the task to require Node.js 16. It previously required 10
```
Comment (zhenlan) [inline releaseNotes/AzureDevOpsPushPipelineExtension.md:6]: This is a breaking change too.

File: releaseNotes/AzureDevOpsPushPipelineExtension.md [hunk 2]
```diff
@@ -3,7 +3,15 @@ Azure App Configuration Push extension for Azure DevOps pipeline can be installe
...
-Updated the task to require Node.js 16. It previously required 10
+* Updated the task to require Node.js 16. It previously required 10
+
+* **Breaking change**: 
+
+    **Before**: 
+      Import feature flags to store by specifying the content type as “application/vnd.microsoft.appconfig.ff+json;charset=utf-8 ” and specifying the prefix as “.appconfig.featureflag/”.
+
+    **After**:
+      Import feature flags to store by specifying the prefix as “.appconfig.featureflag/” is not supported and it will be treated as key-value. To import feature flags please follow [feature management schema](https://github.com/microsoft/FeatureManagement-Dotnet/blob/release/v3/docs/schemas/FeatureManagement.v1.0.0.json).
```
Comment (zhenlan) [inline releaseNotes/AzureDevOpsPushPipelineExtension.md:14]: Is it true? I thought it was prohibited.

File: releaseNotes/AzureDevOpsPushPipelineExtension.md (hunk 1 above)
Comment (MaryanneNjeri) [reply releaseNotes/AzureDevOpsPushPipelineExtension.md]: Yes, updated to document the breaking change.

File: releaseNotes/AzureDevOpsPushPipelineExtension.md (hunk 2 above)
Comment (MaryanneNjeri) [reply releaseNotes/AzureDevOpsPushPipelineExtension.md]: Sorry my bad, it's prohibited,  we throw an error to let user know.

File: releaseNotes/AzureDevOpsPushPipelineExtension.md [hunk 3]
```diff
@@ -3,7 +3,22 @@ Azure App Configuration Push extension for Azure DevOps pipeline can be installe
...
+* **Breaking change**:
+  With the update to the task to require Node.js 16. It previously required 10. The minimum agent version for running the task on self hosted agent changed. 
+
+    **Before**:
+        [Azure Pipeline agent version 2.144.0](https://github.com/microsoft/azure-pipelines-agent/releases/tag/v2.144.0) or later.
+
+    **After**:
+        [Azure Pipeline agent version 2.206.1](https://github.com/microsoft/azure-pipelines-agent/releases/tag/v2.206.1) or later.
+
+* **Breaking change**:
```
Comment (RichardChen820) [inline releaseNotes/AzureDevOpsPushPipelineExtension.md:15]: How about the escaped Key Vault reference json string issue that is reported by the customer? It's another break change

Comment (MaryanneNjeri) [reply releaseNotes/AzureDevOpsPushPipelineExtension.md:15]: Yes, let me make the update

File: releaseNotes/AzureDevOpsPushPipelineExtension.md [hunk 4]
```diff
@@ -3,7 +3,31 @@ Azure App Configuration Push extension for Azure DevOps pipeline can be installe
...
+    **Before**:
+        [Azure Pipeline agent version 2.144.0](https://github.com/microsoft/azure-pipelines-agent/releases/tag/v2.144.0) or later.
+
+    **After**:
+        [Azure Pipeline agent version 2.206.1](https://github.com/microsoft/azure-pipelines-agent/releases/tag/v2.206.1) or later.
+
+* **Breaking change**: 
+
+    **Before**: 
+      Import feature flags to store by specifying the content type as “application/vnd.microsoft.appconfig.ff+json;charset=utf-8 ” and specifying the prefix as “.appconfig.featureflag/”.
```
Comment (RichardChen820) [inline releaseNotes/AzureDevOpsPushPipelineExtension.md:18]: Remove extra whitespace

File: releaseNotes/AzureDevOpsPushPipelineExtension.md [hunk 5]
```diff
@@ -3,7 +3,30 @@ Azure App Configuration Push extension for Azure DevOps pipeline can be installe
...
+* **Breaking change**:
+  With the update to the task to require Node.js 16. It previously required 10. The minimum agent version for running the task on self hosted agent changed.
+
+    **Before**:
+      [Azure Pipeline agent version 2.144.0](https://github.com/microsoft/azure-pipelines-agent/releases/tag/v2.144.0) or later.
+
+    **After**:
+      [Azure Pipeline agent version 2.206.1](https://github.com/microsoft/azure-pipelines-agent/releases/tag/v2.206.1) or later.
+
+* **Breaking change**:
```
Comment (jimmyca15) [inline releaseNotes/AzureDevOpsPushPipelineExtension.md:15]: I'd recommend having a single breaking changes section with bullet points below of the breaking changes.

## Breaking Changes

* Updated the task to require Node.js 16. It previously required 10.
* Updated the minimum supported agent version to 2.206.1. Previously it was 2.144.0

File: releaseNotes/AzureDevOpsPushPipelineExtension.md [hunk 6]
```diff
@@ -3,7 +3,30 @@ Azure App Configuration Push extension for Azure DevOps pipeline can be installe
...
+
+    **Before**:
```
Comment (jimmyca15) [inline releaseNotes/AzureDevOpsPushPipelineExtension.md:17]: The before and after is a bit hard to follow. I'd like if we can be a bit more direct on what's broken since these are often used for reference. It sounds like maybe it should be:

* Feature flag import now requires adherence to the [feature management schema](https://github.com/microsoft/FeatureManagement-Dotnet/blob/release/v3/docs/schemas/FeatureManagement.v1.0.0.json). Settings with feature flag prefix and content type will fail to import.



---

A follow up question here. Why did this break?

Comment (MaryanneNjeri) [reply releaseNotes/AzureDevOpsPushPipelineExtension.md]: Updated to have the Breaking changes with bullet point and removed the before and after.

Comment (MaryanneNjeri) [reply releaseNotes/AzureDevOpsPushPipelineExtension.md]: In version 5 of the Push task, we  integrated with the sync lib, which tries to have the import behavior consistent with the CLI, the CLI logs a warning when importing kvs, and key starts with feature flag prefix, https://github.com/Azure/azure-cli/blob/fde406f3aa731d33e08ef7a77404b25f10f3b559/src/azure-cli/azure/cli/command_modules/appconfig/_kv_helpers.py#L108

File: releaseNotes/AzureDevOpsPushPipelineExtension.md (hunk 5 above)
Comment (MaryanneNjeri) [reply releaseNotes/AzureDevOpsPushPipelineExtension.md]: Updated to have a single breaking changes section with bullet points. 
To confirm for the Pull task since we only document one breaking change ie on the Node js upgrade, we don't need to have it in bullet points and can have the before and after sections?

File: releaseNotes/AzureDevOpsPushPipelineExtension.md (hunk 6 above)
Comment (jimmyca15) [reply releaseNotes/AzureDevOpsPushPipelineExtension.md]: The CLI logs a warning, yet this fails outright? Doesn't seem like consistent behavior.

Comment (MaryanneNjeri) [reply releaseNotes/AzureDevOpsPushPipelineExtension.md]: Since Sync client lib doesn't take a logger object,  all warnings are treated as errors in the Sync lib. Though am thinking maybe the Push task can catch this error and log it as a warning.

Comment (jimmyca15) [reply releaseNotes/AzureDevOpsPushPipelineExtension.md]: The portal doesn't allow creation of feature flags as raw KVs.

Push task doesn't allow it (our current discussion)

but CLI does allow it.

I think first we need to align on what we want the behavior to be across all client side tools.

File: releaseNotes/AzureDevOpsPipelineExtension.md [hunk 7]
```diff
@@ -5,7 +5,15 @@ The Azure App Configuration extension in Azure DevOps pipeline has been publishe
 ![sample](pictures/AzureDevOpsExtensionVersionSample.PNG)
 
 ### v6.0.0 - February, 02 2023
-* Updated the task to require Node.js 16. It previously required 10.
+* **Breaking change**:
+  With the update to the task to require Node.js 16. It previously required 10. The minimum agent version for running the task on self hosted agent changed.
```
Comment (jimmyca15) [inline releaseNotes/AzureDevOpsPipelineExtension.md:9]: This needs the same update as the other file.

Comment (MaryanneNjeri) [reply releaseNotes/AzureDevOpsPipelineExtension.md]: Updated and removed the before/after

File: releaseNotes/AzureDevOpsPushPipelineExtension.md (hunk 6 above)
Comment (MaryanneNjeri) [reply releaseNotes/AzureDevOpsPushPipelineExtension.md]: The Portal supports importing configurations from a file and specifying the prefix as ".appconfig.featureflag/" and content type "application/vnd.microsoft.appconfig.ff+json;charset=utf-8"

File: releaseNotes/AzureDevOpsPipelineExtension.md [hunk 8]
```diff
@@ -5,7 +5,10 @@ The Azure App Configuration extension in Azure DevOps pipeline has been publishe
 ![sample](pictures/AzureDevOpsExtensionVersionSample.PNG)
 
 ### v6.0.0 - February, 02 2023
-* Updated the task to require Node.js 16. It previously required 10.
+* **Breaking changes**:
+  - Updated the task to require Node.js 16. It previously required 10.
+  - Updated the minimum supported azure pipeline agent version to [2.206.1](https://github.com/microsoft/azure-pipelines-agent/releases/tag/v2.206.1) or later. Previously it was [2.144.0](https://github.com/microsoft/azure-pipelines-agent/releases/tag/v2.144.0).
+
 * Fixed a bug that caused an invalid secret URL warning when resolving key vault references in sovereign clouds.
```
Comment (jimmyca15) [inline releaseNotes/AzureDevOpsPipelineExtension.md:12]: Is this a breaking change? If not, a new section header is needed to separate it from **Breaking changes**, such as **Bug fixes**

File: releaseNotes/AzureDevOpsPipelineExtension.md [hunk 9]
```diff
@@ -5,7 +5,10 @@ The Azure App Configuration extension in Azure DevOps pipeline has been publishe
 ![sample](pictures/AzureDevOpsExtensionVersionSample.PNG)
 
 ### v6.0.0 - February, 02 2023
-* Updated the task to require Node.js 16. It previously required 10.
+* **Breaking changes**:
```
Comment (jimmyca15) [inline releaseNotes/AzureDevOpsPipelineExtension.md:8]: Breaking changes shouldn't be a bullet, it should be a header.

File: releaseNotes/AzureDevOpsPipelineExtension.md (hunk 8 above)
Comment (MaryanneNjeri) [reply releaseNotes/AzureDevOpsPipelineExtension.md]: Yes its not a breaking change, added new section header "Bug fixes"

File: releaseNotes/AzureDevOpsPipelineExtension.md (hunk 9 above)
Comment (MaryanneNjeri) [reply releaseNotes/AzureDevOpsPipelineExtension.md]: Fixed breaking change header

File: releaseNotes/AzureDevOpsPushPipelineExtension.md [hunk 10]
```diff
@@ -3,7 +3,11 @@ Azure App Configuration Push extension for Azure DevOps pipeline can be installe
 ![sample](pictures/AzureDevOpsPushExtensionVersionSample.PNG)
 
 ### v5.0.0 - February, 02 2023
-Updated the task to require Node.js 16. It previously required 10
+**Breaking Changes**
+  - Updated the task to require Node.js 16. It previously required 10.
+  - Updated the minimum supported azure pipeline agent version to [2.206.1](https://github.com/microsoft/azure-pipelines-agent/releases/tag/v2.206.1) or later. Previously it was [2.144.0](https://github.com/microsoft/azure-pipelines-agent/releases/tag/v2.144.0).
+  - Importing key vault references value as an escaped json strings is not supported, please specify the key vault reference value as a json object.
```
Comment (jimmyca15) [inline releaseNotes/AzureDevOpsPushPipelineExtension.md:9]: This is a change for all files using application/json content type right, not just kv set? If so, we'll want to call that out.

Also, may help to provide a snippet of what has changed.

Something like

Before when you had this

```
bla bla
```

you got this
```
bla bla bla
```

Now, the input needs to be 
```
bla bla bla
```

Comment (MaryanneNjeri) [reply releaseNotes/AzureDevOpsPushPipelineExtension.md:9]: It only applies for default content profile, using application/json content type. 
How about something like :
"Importing configurations with values as escaped json strings and the content type "application/json" is no longer supported. Please specify value as json object."

**Before**
 Sample configuration file
```
{  
  "app": "{\"uri\":\"https://keyvault.vault.azure.net/secrets/secret\"}"
}
```
Results:

| Key           | Value         |
| ----------- | ----------- |
| app           |   "{"uri":"https://keyvault.vault.azure.net/secrets/secret"}"        |

**After**
The input should be

```
{  
  "app": {
      "uri": "https://keyvault.vault.azure.net/secrets/secret " 
     }
}
```

Comment (MaryanneNjeri) [reply releaseNotes/AzureDevOpsPushPipelineExtension.md:9]: I was thinking since the breaking change is in relation to importing configurations, we can  show the before and after of the snipped that has changed and skip showing the results

Comment (jimmyca15) [reply releaseNotes/AzureDevOpsPushPipelineExtension.md:9]: >How about something like :
"Importing configurations with values as escaped json strings and the content type "application/json" is no longer supported. Please specify value as a json object."


I would avoid the phrase "is no longer supported." It's not that we removed support but rather we changed the behavior. We now convert the values of properties in the settings to json strings.

> I was thinking since the breaking change is in relation to importing configurations, we can show the before and after of the snipped that has changed and skip showing the results

Sounds reasonable. But, I would include the content type that goes with the associated input snippet

Comment (MaryanneNjeri) [reply releaseNotes/AzureDevOpsPushPipelineExtension.md]: > I would avoid the phrase "is no longer supported." It's not that we removed support but rather we changed the behavior. We now convert the values of properties in the settings to json strings.

Yes good point.

Comment (MaryanneNjeri) [reply releaseNotes/AzureDevOpsPushPipelineExtension.md]: This has been updated thanks

File: releaseNotes/AzureDevOpsPushPipelineExtension.md [hunk 11]
```diff
@@ -3,7 +3,43 @@ Azure App Configuration Push extension for Azure DevOps pipeline can be installe
...
+    task: AzureAppConfigurationPush@5
+    inputs:
+      azureSubscription: '<subscription>'
+      AppConfigurationEndpoint: '<store endpoint>'
+      ....
+      ContentType: 'application/vnd.microsoft.appconfig.keyvaultref+json;charset=utf-8'
+
+    ```
+
+  - Feature flag import now requires adherence to the [feature management schema](https://github.com/microsoft/FeatureManagement-Dotnet/blob/release/v3/docs/schemas/FeatureManagement.v1.0.0.json). Settings with feature flag prefix and content type will fail to import
```
Comment (bbrandt) [inline releaseNotes/AzureDevOpsPushPipelineExtension.md:42]: Does specifying the feature flags as the [feature management schema](https://github.com/microsoft/FeatureManagement-Dotnet/blob/release/v3/docs/schemas/FeatureManagement.v1.0.0.json) for import then get converted to the App Config service prefix and content type for feature flags by the extension or the app config service?  So that the imported feature flags appear in the Feature Manager view alongside feature flags manually created in the portal?

File: releaseNotes/AzureDevOpsPushPipelineExtension.md [hunk 12]
```diff
@@ -3,7 +3,43 @@ Azure App Configuration Push extension for Azure DevOps pipeline can be installe
 ![sample](pictures/AzureDevOpsPushExtensionVersionSample.PNG)
 
 ### v5.0.0 - February, 02 2023
-Updated the task to require Node.js 16. It previously required 10
+**Breaking Changes**
+  - Updated the task to require Node.js 16. It previously required 10.
+  - Updated the minimum supported azure pipeline agent version to [2.206.1](https://github.com/microsoft/azure-pipelines-agent/releases/tag/v2.206.1) or later. Previously it was [2.144.0](https://github.com/microsoft/azure-pipelines-agent/releases/tag/v2.144.0).
+  - Importing configurations with values as escaped JSON strings and the content type `application/json`, introduces a modified behavior, the values of properties in the settings are expected to be specified as JSON objects. Please ensure the values are JSON objects.
```
Comment (jimmyca15) [inline releaseNotes/AzureDevOpsPushPipelineExtension.md:9]:
```suggestion
  - The behavior when importing configurations with JSON content types such as `application/json` or `application/vnd.mycustomresource+json` has changed. The properties in the provided configuration are serialized as JSON to respect the provided content type. Settings that are expected to end up as JSON serialized strings in App Configuration should now be specified as JSON objects.

```

**Review by jimmyca15 — APPROVED** [review]
